app/grandchallenge/serving/views.py

```diff
@@ -131,3 +132,19 @@ def _create_download(*, creator_id, image_id=None, submission_id=None):
 
     if n_updated == 0:
         Download.objects.create(**kwargs)
+
+
+def serve_structured_challenge_submission_form(
+    request, *, challenge_request_pk, **_
+):
+    try:
+        challenge_request = ChallengeRequest.objects.get(
+            pk=challenge_request_pk
+        )
+    except ChallengeRequest.DoesNotExist:
+        raise Http404("Challenge request not found.")
+
+    if request.user.has_perm("view_challengerequest"):
+        return protected_storage_redirect(
+            name=challenge_request.structured_challenge_submission_form.name
+        )
```

Comment:
A functional view should never return None (implicitly done here). You need to raise `PermissionDenied` if the user cannot access the file.

Reply:
This method also needs tests.